core/templates/dev/head/pages/exploration_player/PlayerTranscriptService.js

```diff
@@ -20,146 +20,146 @@
+          stateName, params, contentHtml, interactionHtml, leadsToConceptCard) {
+        transcript.push(
+          StateCardObjectFactory.createNewCard(
+            stateName, params, contentHtml, interactionHtml, leadsToConceptCard
+          )
+        );
+        numAnswersSubmitted = 0;
+      },
+      addPreviousCard: function() {
+        if (transcript.length === 1) {
+          throw Error(
+            'Exploration player is on the first card and hence no previous ' +
+            'card exists.');
+        }
+        // TODO(aks681): Once worked examples are introduced, modify the below
+        // line to take into account the number of worked examples displayed.
+        var previousCard = angular.copy(transcript[transcript.length - 2]);
+        previousCard.setLeadsToConceptCard(false);
+        transcript.push(previousCard);
+      },
+      setDestination: function(newDestStateName) {
+        var lastCard = this.getLastCard();
+        if (lastCard.getDestStateName()) {
+          throw Error(
+            'Trying to set a destStateName when it has already been set.',
+            transcript);
+        }
 
-      lastCard.destStateName = newDestStateName;
-    },
-    addNewInput: function(input, isHint) {
-      var pairs = transcript[transcript.length - 1].inputResponsePairs;
-      if (pairs.length > 0 &&
-          pairs[pairs.length - 1].oppiaResponse === null) {
-        throw Error(
-          'Trying to add an input before the response for the previous ' +
-          'input has been received.',
-          transcript);
-      }
-      if (!isHint) {
-        numAnswersSubmitted += 1;
-      }
-      transcript[transcript.length - 1].inputResponsePairs.push({
-        learnerInput: input,
-        oppiaResponse: null,
-        isHint: isHint
-      });
-    },
-    addNewResponse: function(response) {
-      var pairs = transcript[transcript.length - 1].inputResponsePairs;
-      if (pairs[pairs.length - 1].oppiaResponse !== null) {
-        throw Error(
-          'Trying to add a response when it has already been added.',
-          transcript);
-      }
-      pairs[pairs.length - 1].oppiaResponse = response;
-    },
-    getNumCards: function() {
-      return transcript.length;
-    },
-    getCard: function(index) {
-      if (index < 0 || index >= transcript.length) {
-        $log.error(
-          'Requested card with index ' + index +
-          ', but transcript only has length ' + transcript.length + ' cards.');
-      }
-      return transcript[index];
-    },
-    getLastAnswerOnActiveCard: function(index) {
-      if (this.isLastCard(index) || transcript[index].stateName === null) {
-        return null;
-      } else {
-        return transcript[index].
-          inputResponsePairs.slice(-1)[0].learnerInput;
+        lastCard.setDestStateName(newDestStateName);
+      },
+      addNewInput: function(input, isHint) {
+        var pairs = transcript[transcript.length - 1].getInputResponsePairs();
```

Comment:
Perhaps worth defining and using a helper getLastCard() method here, and a getLastInputResponsePair() method on the card object?

Reply:
Done, getLastCard() was already there, used that, and added getLastOppiaResposne().